### DISCUSSION TITLE
Is it possible to customize the install location of `micropip`?

### DISCUSSION THREAD

**Discussion by CNSeniorious000** [pyodide/pyodide#4985] [General]
I am trying to implement isolated workspaces with a shared pyodide instance.

The difficulty is handling different contexts. For now I can handle the working directory, project files, `sys.path` well, and even imported modules, but I can't handle 3rd packages well.

So I am considering, can there be a way to customize the install path of `micropip`? Like, I'd like to install it to a local dir like `.venv` (of cource in the virtual fs of Emscripten).

Or, if this is hard to implement, I wonder are there some elegant way to do this? Currently I can only think of this way:

1. Take a snapshot of the tree structure of the global `site-packages` dir when entering a new workspace
2. Invoke `micropip.install`
3. Take another snapshot, calculate the increment
4. When closing this workspace, cleanup all these files and delete their keys from the `sys.modules`

**Comment by ryanking13** [comment]
I think it is feasible. We may support it in an ad-hoc way. However, to support this more precisely, I would like the following tasks to be done first.

1. https://github.com/pyodide/micropip/issues/112 would allow each `micropip` instance to have different installation directories.
2. https://github.com/pyodide/pyodide/issues/4914 would allow syncing `pyodide.loadPackage` and `micropip.install` to share the same installation mechanism.
